fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/CursorLimitManager.java

```diff
@@ -136,7 +139,8 @@ public boolean tryRecordScan() {
         if (!halted) {
             usedInitialPass = true;
         } else if (failOnScanLimitReached) {
-            throw new ScanLimitReachedException("limit on number of key-values scanned per transaction reached");
+            throw new ScanLimitReachedException("limit on number of key-values scanned per transaction reached")
+                    .addLogInfo("no_next_reason", getStoppedReason().map(reason -> reason.toString()).orElse("Unknown."));
```

Comment:
As a note, there should be no reason for the `.map()`.  The logging info allows for you to log any kind of object and it will automatically have `.toString()` called on it when logged, so:

    getStoppedRead().orElse("Unknown")

would be fine.